.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,61 @@
+name: Unit tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  swiftlint:
+
+    name: Run SwiftLint
+
+    if: github.event_name == 'pull_request'
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out the code
+      uses: actions/checkout@v3
+    - name: Fetch PR's target branch
+      run: git fetch --no-tags --prune --depth=1 origin ${{ github.base_ref }}
+    - name: Run SwiftLint on files changed in the PR
```

Comment:
We've recently changed this in the iOS repo to be called for all files, not only those changed in the PR - see [this task](https://app.asana.com/0/0/1202474746199344). macOS is still not updated. From what I checked, there are 6 violations currently in TRK, all should be easily fixable. Would you mind fixing them and updating this step to check all files on every run? (see [iOS workflow](https://github.com/duckduckgo/iOS/blob/develop/.github/workflows/pr.yml#L17) for reference) - we could then also enable strict checking.

When this is updated, we could remove swiftlint.yml, because checks in this file are run on pushes to main too.